Making a style guide

ACharbonneau opened the pull request:
@marisalim 

This is not remotely ready to merge, but I wanted to give you more of a framework for what I was asking for. The people we are expecting to contribute to the UCL are not just our team, and not necessarily even just from the CFDE. It needs *everything* explained. What files they need, in what format, naming conventions, how we want PRs. I've left a lot of ellipses for you to fill in. 

What might help for you to get an idea what I'm looking for, is for you to actually just write a small use case, and see what things you have to go dig around to see how they should be. We don't want to make our users have to go dig around, we want everything they need to know in one place. 


Preview: https://cfde-usecases.readthedocs-hosted.com/en/styleguide/about/contributing/

ACharbonneau committed:
Making a style guide

marisalim commented:
@ACharbonneau what do the green check marks mean?
![image](https://user-images.githubusercontent.com/5659802/88347880-bc3c8b80-cd00-11ea-848c-097e01b1b5fd.png)

ACharbonneau commented:
Those are things that are done. So. As part of the July demo, we met all of the checked requirements, and demonstrated those two use cases. So, people should *not* add check marks when they are submitting things. The Use Case committee (which is currently me) will give use cases the correct marks for done, approved, and in progress as they are appropriate

marisalim committed:
adding style guide content

add - file format and pg section requirements, add content PR format guide

marisalim committed:
define yml parent types

marisalim committed:
parent is tab name

marisalim committed:
add md syntax link

marisalim commented:
@ACharbonneau will the user task and requirement pages eventually need descriptions? right now they have links to which use case and/or user task they appear in.

marisalim committed:
test fixing bullet pt format

marisalim committed:
backticks render weird - testing

marisalim committed:
testing code block

marisalim committed:
a use case file template

marisalim committed:
persona file template

marisalim committed:
obj file template

marisalim committed:
user task template file

marisalim committed:
requirement template file

marisalim committed:
test template links

these links will have to be edited after merge, they are links on this branch

marisalim committed:
edit template_files path

ACharbonneau commented:
The requirements and user tasks are supposed to be short and self explanatory enough that you know what they are from the title. However, if someone who isn't familiar with the library doesn't know what they mean, then sure, we can add descriptions. What do you think?

marisalim committed:
note about user task, reqs titles

should be self-expl, but add description as needed

marisalim committed:
word smithing

marisalim committed:
format, word smith edits

marisalim committed:
bullet list spacing

ACharbonneau commented on docs/about/contributing.md:
```diff
@@ -30,19 +30,16 @@ If you are familiar with GitHub please either:
 ### Use Case approval process
 - The Use Case committee will mark proposed use cases as `in progress`, `approved`, and `done`, as appropriate.
-  
 - If the Use Case has been approved, a &#x2705; will be added next to the Use Case tab title, as well as its corresponding Requirements tab titles.
```
A checkmark means done, not approved. We need to find approved and in progress insignia

marisalim replied:
Ah ok, will edit!

marisalim replied:
how about this?
:hourglass_flowing_sand: for in progress (HEX: 23F3)
:ok_hand: for approved (HEX: 1F44C)

ACharbonneau replied:
Works for me

marisalim committed:
drafting approval symbols, formatting

marisalim committed:
formatting, note on pg num & file IDs

marisalim committed:
Merge branch 'master' into StyleGuide

marisalim committed:
description edit

marisalim committed:
persona file description edit

marisalim commented:
@ACharbonneau I'm at a point where I need feedback! I think I can't add you as a reviewer because you made this PR :P

ACharbonneau committed:
Update contributing.md

ACharbonneau commented on docs/about/contributing.md:
```diff
@@ -7,21 +7,139 @@ nav_order: 2
+### `Persona` files
+- Required sections:
+    - yaml index header
+    - One line description of the biological/computational experience and relation to any of the other Personas in the Use Cases Repository
```
I honestly have no idea why I made two description sections in these...can you just merge the two descriptions together and take out the extra header line

marisalim replied:
done! also went ahead and edited out the extra header lines in the existing persona files

ACharbonneau commented on docs/about/contributing.md:
```diff
@@ -7,21 +7,139 @@ nav_order: 2
+  - or write up your use case and submit it as a pull request (PR) at [https://github.com/nih-cfde/usecases](https://github.com/nih-cfde/usecases). Please follow the [Use Case Style Guide](#usecasestyle) below.
+
+### PR format
+If you are submitting a pull request:
```
Let's make an actual PR template and have the instructions here tell them to fill it in completely

marisalim replied:
i drafted the steps for PR to preview and then PR to master... not sure if this is what we want ppl to do, please edit!

ACharbonneau commented on docs/about/contributing.md:
```diff
@@ -7,21 +7,139 @@ nav_order: 2
+
+If you are familiar with GitHub please either:
+
+  - [open an issue](https://github.com/nih-cfde/usecases/issues/new) describing your idea
```
Lets make a new use case suggestion issue template

ACharbonneau commented:
This looks really good! I love the idea of having templates for the different docs. I left a few comments. See if you can get the issue and PR templates working, poke me with questions if you have them.

- [instructions](https://docs.github.com/en/github/building-a-strong-community/configuring-issue-templates-for-your-repository)
- [example](https://github.com/nih-cfde/deliverables/tree/master/.github/ISSUE_TEMPLATE)

ACharbonneau committed:
Update objective-template.md

marisalim committed:
merge description section template

marisalim committed:
persona description edit

marisalim committed:
remove extra header

marisalim committed:
remove extra header

marisalim committed:
remove extra header

marisalim committed:
remove extra header

marisalim committed:
remove extra header

marisalim committed:
remove extra header

marisalim committed:
testing PR template

marisalim committed:
testing issue template

marisalim committed:
issue template layout box

marisalim committed:
fix temp file paths, PR instructions

rel paths instead of web link
PR for preview, PR to master

doesn't have issue or PR template paths because i don't know how to link them yet

marisalim committed:
fix broken links

marisalim committed:
bullet list fix format

marisalim committed:
urg try PR list again

ACharbonneau commented on docs/about/contributing.md:
```diff
@@ -32,15 +32,15 @@ If you are submitting a pull request, please create one pull request per use cas
+  - If the checks complete successfully, merge the changes and view the [preview website](https://cfde-usecases.readthedocs-hosted.com/en/preview/) to check the changes. It can take ~30 minutes for the new changes to appear. If the checks failed, you can submit an issue for help.
 
 2. If you are satisfied with the changes, submit a new PR from your branch to the `master` branch.
-- See [template]()
```
You don't need to tell them about the template like this. When they submit the PR it will pop up. Just merge this and see what it does, and then it will be easier for you to write this.

marisalim committed:
take out pr template link

marisalim reviewed (approved):
merge to check PR and issue templates
